src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildANetCoreApp.cs

```diff
@@ -925,5 +925,113 @@ public void It_builds_the_project_successfully_with_only_reference_assembly_set(
+        [InlineData(ToolsetInfo.CurrentTargetFramework, new[] { "ubuntu.22.04-x64" }, true, true, null, true)]
+        [InlineData(ToolsetInfo.CurrentTargetFramework, new[] { "ubuntu.22.04-x64" }, true, false, null, true)]
+        [InlineData(ToolsetInfo.CurrentTargetFramework, new[] { "ubuntu.22.04-x64" }, false, true, null, true)]
+        // Non-portable and portable RIDs should warn
+        [InlineData(ToolsetInfo.CurrentTargetFramework, new[] { "ubuntu.22.04-x64", "win7-x86", "unix" }, true, true, null, true)]
+        // Portable RIDs only should not warn
+        [InlineData(ToolsetInfo.CurrentTargetFramework, new[] { "win-x86", "linux", "linux-musl-x64", "osx-arm64", "unix" }, true, true, null, false)]
+        // No RID assets should not warn
+        [InlineData(ToolsetInfo.CurrentTargetFramework, new string[] { }, false, false, null, false)]
+        // Below .NET 8 should not warn
+        [InlineData("net7.0", new string[] { "ubuntu.22.04-x64", "win7-x86" }, true, true, null, false)]
+        // Explicitly set to use RID graph should not warn
+        [InlineData(ToolsetInfo.CurrentTargetFramework, new[] { "alpine-x64" }, true, true, true, false)]
+        // Explicitly set to not use RID graph should warn
+        [InlineData(ToolsetInfo.CurrentTargetFramework, new[] { "alpine-x64" }, true, true, false, true)]
+        public void It_warns_on_nonportable_rids(string targetFramework, string[] rids, bool addLibAssets, bool addNativeAssets, bool? useRidGraph, bool shouldWarn)
+        {
+            var packageProject = new TestProject()
+            {
+                Name = "WithRidAssets",
+                TargetFrameworks = targetFramework,
+            };
+
+            // Add assets for each RID. The test just needs the asset to exist, so it can just copy same output assembly.
+            foreach (string rid in rids)
+            {
+                if (addLibAssets)
+                {
+                    packageProject.AddItem("None",
+                        new Dictionary<string, string>()
+                        {
+                        { "Include", $"$(TargetPath)" },
+                        { "Pack", "true" },
+                        { "PackagePath", $@"runtimes\{rid}\lib\$(TargetFramework)" }
+                        });
+                }
+
+                if (addNativeAssets)
+                {
+                    packageProject.AddItem("None",
+                        new Dictionary<string, string>()
+                        {
+                            { "Include", $"$(TargetPath)" },
+                            { "Pack", "true" },
+                            { "PackagePath", $@"runtimes\{rid}\native" }
+                        });
+                }
+            }
+
+            TestAsset packageAsset = null;
+            TestAsset testAsset = null;
+            try
+            {
+                packageAsset = _testAssetsManager.CreateTestProject(packageProject);
```

Comment:
if you set the `identifier` parameter on this call to something unique for each set of test input parameters (this is commonly just concatting them) then the test framework will give you unique test folders (as well as clean them up). This should prevent the kind of manual bookkeeping you're doing at the end of this test method.

Reply:
Ah, thanks.